src/main/clojure/powderkeg/core.clj

```diff
@@ -247,7 +247,7 @@
         df (barrier-fn (fn [it] (eduction (comp may-unmap-tuple2 xform may-map-tuple2) (iterator-seq it))))
         rdd (.mapPartitions rdd
               (reify org.apache.spark.api.java.function.FlatMapFunction ; todo: skip api.java.* go to spark
-                (call [_ it] ((df) it)))
+                (call [_ it] (.iterator ((df) it))))
```

Comment:
Hmm, seems so https://spark.apache.org/docs/2.1.0/api/java/org/apache/spark/api/java/function/FlatMapFunction.html vs 1.5.0. Should we do a different change here? Above only avoids ClassCastException.